### DISCUSSION TITLE
Creating multiple participants in the same application

### DISCUSSION THREAD

**Discussion by NickBarrettHFX** [eProsima/Fast-DDS#3272] [Unofficial support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

I expect to be able to create multiple dds participants within the same application on the same domain. 

### Current behavior

My application is crashing with the following error

[foonathan::memory] Buffer overflow at address 0x56325d56d380 detected, corresponding memory block 0x56325d56d350 has only size 48.[foonathan::memory] Buffer overflow at address 0x7f2d90058fa0 detected, corresponding memory block 0x7f2d90058f70 has only size 48.Aborted

Crashing leaves some blocked files in my /dev/shm directory which I then manually delete. 

### Steps to reproduce

- Create two DomainParticipantQos objects, with different names.
- Use the DomainParticipantFactoryClass to create a participant using the first QoS object (participant-1)
- Use the DomainParticipantFactoryClass to create another participant (participant-2)
- Delete participant-1
- Delete participant-2

I have also tried creating and deleting the particpants in their own thread, to no avail. See code below:

```
auto t = std::thread([](){
  eprosima::fastdds::dds::DomainParticipantQos TestParticipantQos;
  TestParticipantQos.name("test_participant");

  eprosima::fastdds::dds::DomainParticipant* dds_participant_1 =
      eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->create_participant(0, TestParticipantQos);

      std::this_thread::sleep_for(std::chrono::seconds(5));

  if (eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->delete_participant(dds_participant_1) !=
      ReturnCode_t::RETCODE_OK) {
    // 
  }

});

auto t2 = std::thread([](){
  eprosima::fastdds::dds::DomainParticipantQos TestParticipant2Qos;
  TestParticipant2Qos.name("test_participant_2");

  eprosima::fastdds::dds::DomainParticipant* dds_participant_2 =
      eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->create_participant(0, TestParticipant2Qos);

  std::this_thread::sleep_for(std::chrono::seconds(5));

  if (eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->delete_participant(dds_participant_2) !=
      ReturnCode_t::RETCODE_OK) {
    // 
  }
});

t.join();
t2.join();
```

### Fast DDS version/commit

Version 2.3.3

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Shared Memory Transport (SHM)

### Additional context

Running in a docker container with Linux, gcc7, qt5.12

### Relevant log output

```shell
2022-12-02 14:14:53.995 [RTPS_TRANSPORT_SHM Warning] (ID:140546597336960) Port 7412 Zombie. Reset the port -> Function open_port_internal
2022-12-02 14:14:54.018 [RTPS_TRANSPORT_SHM Warning] (ID:140546597336960) Port 7413 Zombie. Reset the port -> Function open_port_internal
2022-12-02 14:14:54.026 [RTPS_TRANSPORT_SHM Warning] Buffer is being invalidated, segment_size may be insufficient -> Function invalidate_if_not_processing
2022-12-02 14:14:54.027 [RTPS_TRANSPORT_SHM Warning] Buffer is being invalidated, segment_size may be insufficient -> Function invalidate_if_not_processing
[foonathan::memory] Buffer overflow at address 0x55f4e3e0c5f0 detected, corresponding memory block 0x55f4e3e0c5c0 has only size 48.2022-12-02 14:14:59.024 [RTPS_READER Warning] Automatically unmatching on ~WriterProxy -> Function ~WriterProxy
Aborted
```

**Comment by Mario-DL** [comment]
Hi @NickBarrettHFX,

According to our CONTRIBUTING.md guidelines, gcc 7 is not an officially supported compiler (gcc 9 is the minimum required), so I am moving this issue to the corresponding discussion forum.
